ncm-ssh/src/main/perl/ssh.pm

```diff
@@ -20,20 +20,24 @@ our $NoActionSupported = 1;
 
 use CAF::Process;
 use CAF::FileEditor;
-use constant SSHD_CONFIG  => "/etc/ssh/sshd_config";
-use constant SSH_CONFIG   => "/etc/ssh/ssh_config";
-use constant SSH_VALIDATE => qw(/usr/sbin/sshd -t -f /proc/self/fd/0);
+use constant DEFAULT_SSHD_CONFIG  => "/etc/ssh/sshd_config";
+use constant DEFAULT_SSHD_PATH    => "/usr/sbin/sshd";
+use constant DEFAULT_SSH_CONFIG   => "/etc/ssh/ssh_config";
 
 # Returns true if $file is a valid SSHD configuration file.
 sub valid_sshd_file
 {
-    my ($self, $file) = @_;
+    my ($self, $file, $cfg) = @_;
+
+    # Use /dev/stdin, instead of /proc/self/fd/0 (which is used in some other components).
+    # This is because /proc/self/fd/0 does not exist on Solaris. 
+    my @cmdline = ( $cfg->{sshd_path} || DEFAULT_SSHD_PATH, '-t', '-f', '/dev/stdin' );
```

Comment:
use arayref directly `$cmdline = [...]` and then `CAF::Process->new($cmdline, ...)`